rfc/0011-value-transformer/README.md

```diff
@@ -0,0 +1,191 @@
+
+## Explanation
+
+```
+transformer <name> {
+  input <inputName> oftype <valuetype>; // potentially many
+  output <outputName> oftype <valuetype>; // for now: exactly one
+
+  <outputName>: <expression producing output valuetype>;
+}
+```
+
+A transformer can map none, one, or multiple inputs to at least one output with their types.
+See future enhancements for multiple outputs.
+
+The output value is assigned (`:`) to an expression to produces the output values.
+
+### Example 1: Simple numeric transformation
+```
+transformer CelsiusToKelvin {
+  input tempCelsius oftype decimal;
+  output tempKelvin oftype decimal;
+
+  tempKelvin: tempCelsius + 273.15; // simple calculation
+}
+```
+
+### Example 2: Multiple inputs
+```
+// "MyStreet" and "7b" => "MyStreet 7b"
+transformer AddressComposer {
+  input streetName oftype text;
+  input houseNumber oftype text;
+  output address oftype text;
+
+  address: streetName + " " + houseNumber; // assumption: "+" is string appending (separate RFC)
+}
+```
+
+
+### Example 3: Multiple outputs
+```
+// 456 => (ones:6, tens:5, hundreds:4)
+transformer DigitSeparator {
+  input number oftype integer;
+  output ones oftype integer;
+  output tens oftype integer;
+  output hundreds oftype integer;
+
+  ones: number % 10;
+  tens: floor (number / 10) % 10;
+  hundreds: floor (number / 100);
+}
+```
+
+### Usage 1: TableInterpreter
+
+The `TableInterpreter` block uses a transformer from `text` to the desired value type with the `with` keyword. Parsing can still be used as currently (`as` keyword)
+
+Example:
+```
+block GasReserveTableInterpreter oftype TableInterpreter {
+  header: true;
+  columns: [ // only allow transformers with one input and one output or plain parsing here
+      "Datum" with DateTransformer,
+      "Kritisch" with PercentTransformer,
+      "Angespannt" as text,
+      "Stabil" with GermanToBooleanTransformer,
+      "Speicherstand IST" with PercentTransformer,
+      "gesetzliche Ziele" with PercentTransformer
+    ];
+}
+```
+
+This version only works with transformers that have exactly one input and exactly one output. We might need to come up with a different mechanism to allow other transformers (requires future RFCs).
+
+
+### Usage 2: TableTransformerBlock
+There is a newly introduced `TableTransformerBlock` block type that applies the mapper to a column in a `table`.
+
+Example:
+```
+block GermanToBooleanBlock oftype TableTransformerBlock {
+  inputColumns: ["Kritisch"];  // array to satisfy all inputs of transformer
+  outputColumns: ["IsCritical"]; // might also overwrite one of the input columns
+  use: GermanToBooleanTransformer;
+}
+```
+
+### Usage 3: SheetTransformerBlock
+There is a newly introduced `SheetTransformerBlock` block type that applies the mapper to a set of cells in a `sheet`. Please note that the value type is not stored in sheets and thus lost after computation. Outputs are therefore parsed to string automatically by the block.
```

Comment:
I think this shows a difference in the semantics. One is changing valuetypes into other valuetypes, one is changing the structure of a string. If the valuetype is lost after the transformation, I do not think we should use transformators for it, it should be a different concept that can only map from string to string (?).

Reply:
Or allow only writing to string? Then the user would need to chain a default transformer to convert back to string.